kickstarter/android-oss
# KSCurrency refactor

# What ❓
~First~ Third pass of fixing currency conversions and updating tests:

- I added a Model for `Country` based off https://github.com/kickstarter/ios-oss/blob/master/KsApi/models/Project.Country.swift.
- The method `formatWithProjectCurrency` was redundant because that's already what `format` does. The method `userCurrencyOptions` was also redundant because we can just use `currencyOptions`.
- Almost all of the tests were either incomplete or just testing the wrong thing.
- Most notably all of the tests for `formatWithUserPreference` weren't actually testing anything because the `fxRate` was never updated. 
- When testing conversions, the config `KSCurrency` uses needs to be explicitly set or it defaults to `server-config.json` (which has a user in the US). (Relevant to the `RewardViewHolderViewModel` tests).
- For some reason (it predates me) we were using `float` to represent currency values. iOS uses `double` so I switched to `double` so we can be wrong together.

# How to QA? 🤔
Change your currency preferences and verify the converted amounts are displayed properly.
Friendly reminder, we round ~UP~ HALF_UP (like normal rounding) for reward and checkout amounts and DOWN for pledged and goal amounts.
